### DISCUSSION TITLE
AWS S3 using default credentials file still asks for secret access key

### DISCUSSION THREAD

**Discussion by cal-ellevation** [iterate-ch/cyberduck#16410] [Q&A]
This ticket system is to report bugs and feature requests. For support, visit the [help page](https://cyberduck.io/help/) first.

 * If you have trouble connecting to a server or your login credentials are not valid, try to resolve the issue with the assistance of your hosting service provider. Any such ticket will be closed with a resolution of _thirdparty_.

**Describe the bug**
On MacOS when using an s3 connection through a bookmark without defining access key or secret, it does grab the key from the credentials file, but not the secret. Is that intentional or could it be fixed?

**To Reproduce**
1. Create valid credentials file in ~/.aws/credentials with a default profile.
2. Add a bookmark to cyberduck with empty access key id and secret access key.
3. Attempt to connect to bookmark and see that it is asking for a secret access key even though it's in the profile.

**Expected behavior**
Secret access key is picked up from the credentials file

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS 14.6.1
 - Version 9.0.0 (41777

**Log Files**
Log Files don't show anything related to this.

**Additional context**

**Comment by dkocher** [comment]
According to the documentation [^1] this should work. As a workaround edit the hostname of the bookmark in _Server_ to preset the default credentials from the `~/.aws/credentials` default profile.

[^1]: https://docs.cyberduck.io/protocols/s3/#read-credentials-from-aws-credentials

**Comment by dkocher** [comment]
We are working on a fix in [^1].

[^1]: #16421
